File: socket/ex1/sever.cpp
#define _WINSOCK_DEPRECATED_NO_WARNINGS

#include <WinSock2.h>
#include <stdio.h>
#include <stdlib.h>

#pragma comment(lib, "ws2_32.lib")

int main()
{
    WSADATA wsaData;
    int port = 5099;

    char buf[] = "Server: hello, I am a server.....";

    if (WSAStartup(MAKEWORD(2, 2), &wsaData) != 0)
    {
        printf("Failed to load Winsock");
        return 0;
    }

    //创建用于监听的套接字
    SOCKET sockSrv = socket(AF_INET, SOCK_STREAM, 0);

    SOCKADDR_IN addrSrv;
    addrSrv.sin_family = AF_INET;
    addrSrv.sin_port = htons(port); //1024以上的端口号
    addrSrv.sin_addr.S_un.S_addr = htonl(INADDR_ANY);

    int retVal = bind(sockSrv, (LPSOCKADDR)&addrSrv, sizeof(SOCKADDR_IN));
    if (retVal == SOCKET_ERROR){
        printf("Failed bind:%d\n", WSAGetLastError());
        return 0;
    }

    if (listen(sockSrv, 10) == SOCKET_ERROR){
        printf("Listen failed:%d", WSAGetLastError());
        return 0;
    }

    SOCKADDR_IN addrClient;
    int len = sizeof(SOCKADDR);

//等待客户请求到来    
        SOCKET sockConn = accept(sockSrv, (SOCKADDR *)&addrClient, &len);
        if (sockConn == SOCKET_ERROR){
            printf("Accept failed:%d", WSAGetLastError());
            //break;
        }

        printf("Accept client IP:[%s]\n", inet_ntoa(addrClient.sin_addr));

        //发送数据
        int iSend = send(sockConn, buf, sizeof(buf), 0);
        if (iSend == SOCKET_ERROR){
            printf("send failed");
           // break;
        }

        char recvBuf[100];
        memset(recvBuf, 0, sizeof(recvBuf));
        //         //接收数据
        recv(sockConn, recvBuf, sizeof(recvBuf), 0);
        printf("%s\n", recvBuf);

        closesocket(sockConn);


    closesocket(sockSrv);
    WSACleanup();
    system("pause");
}

File: NetworkExperiment/LocaLoopback/sever.py
import socket

# 创建socket
sever = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # SOCK_STREAM=》TCP协议

# 绑定地址与端口
sever.bind(("127.0.0.1", 8080))  # 本地回环

# 开始监听
sever.listen(100)
print('服务器开始监听' + ' ''服务器IP与端口： %s:%s' % ("127.0.0.1", 8080))

# 当有连接接入时
while True:
    client, client_addr = sever.accept()
    print(str(client_addr) + '连接到了服务器')

    # 与当前客户端进行消息通信循环
    while True:
        data = client.recv(1024)  # 最大接收的字节个数
        if(data == '！'):
            break
        print("收到的客户端数据：", data.decode('utf-8'))
        client.send(data)

    # 当前客户端断开连接
    print(str(client_addr) + '断开连接')
    client.close()

# 关闭
sever.close()
